appletree/component.py

```diff
@@ -147,6 +149,12 @@ def implement_binning(self, mc, eff):
             raise ValueError(f"Unsupported bins_type {self.bins_type}!")
         if self.add_eps_to_hist:
             # as an uncertainty to prevent blowing up
+            if self.bins_type != "meshgrid":
```

Comment:
Since implement_binning will be called many many times during the fitting, maybe this warning can go to somewhere else like __init__?